### PR TITLE
add monotonic_count support

### PR DESCRIPTION
**Pull request by arbll**
### What does this PR do

Adds the ability to jmxfetch to consume "monotonic_counts". This new metric type for jmxfetch is going to be almost identical to the agent checks `monotonic_count`:

> Submit the sampled raw value of your counter. Don’t normalize the values to a rate, or calculate the deltas before submitting. If the value of your counter ever decreases between submissions the resulting stored value for that submission is 0
https://docs.datadoghq.com/developers/metrics/counts/

The only difference with agent checks is that the value will be stored as a `rate` in the backend. Functionality wise this should not have any impact.

This will allow jmxfetch to monitor counters that increase over time like the total number of request received by a web server and most importantly store them as counts in Datadog.

### Notes

This is based on a POC from @mirkoprescha  that you can find here: https://github.com/DataDog/jmxfetch/pull/190.

### CODE REVIEW

File: src/main/java/org/datadog/jmxfetch/reporter/Reporter.java [hunk 1]
```diff
@@ -72,6 +84,28 @@ public void sendMetrics(LinkedList<HashMap<String, Object>> metrics, String inst
...
+                String key = generateId(metric);
+                if (!instanceCountersAggregator.containsKey(key)) {
+                    instanceCountersAggregator.put(key,  currentValue.longValue());
+                    continue;
+                }
+
+                long oldValue = instanceCountersAggregator.get(key);
+                long delta = currentValue.longValue() - oldValue;
+
+                if(Double.isNaN(delta) || Double.isInfinite(delta)) {
```
[the remark this answers is not in the document]
Comment (arbll) [reply src/main/java/org/datadog/jmxfetch/reporter/Reporter.java]: I would guess we could get `NaN` or infinities as input. But you're right I don't think this check matters that much here.

**Comment by l0k0ms** [comment]
I think we should update the java integration here: https://docs.datadoghq.com/integrations/java/#the-attribute-filter. Just open an issue on the doc github repo or in our trello board for tracking and we will do it.

**Comment by truthbk** [comment]
Force-pushed after rebasing on top of `master` following the merge of #98, also fixed a couple of stylistic issues.

File: src/main/java/org/datadog/jmxfetch/reporter/StatsdReporter.java [hunk 2]
```diff
@@ -31,7 +31,9 @@ protected void sendMetricPoint(String metricType, String metricName, double valu
             this.statsDClient.stop();
             init();
         }
-        if (metricType.equals("histogram")) {
+        if (metricType.equals("monotonic_count")) {
+            statsDClient.count(metricName, (long) value, tags);
```
[the remark this answers is not in the document]
Comment (arbll) [reply src/main/java/org/datadog/jmxfetch/reporter/StatsdReporter.java]: `.count()` takes an integer (long) while the others take a real (double)

[the remark this answers is not in the document]
Comment (arbll) [reply src/main/java/org/datadog/jmxfetch/reporter/StatsdReporter.java]: I could refactor things a bit if needed

**Pull request merged**